code/script.js

```diff
@@ -262,82 +417,163 @@ const selectQuestion = () => {
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
 
+  const value = questions.value
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
+      value,
       // 👆 add the value from the input here
-      category: category,
+      category
     }
   } else if (category === 'eye color') {
     // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value,
+      category
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
+      attribute: value,
       // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      category
     }
   } else if (category === 'other') {
     // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category
+   }
   }
+  numberOfGuesses++
+  totalGuesses.innerText = numberOfGuesses
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  const keep = currentQuestion.value === secret[currentQuestion.attribute]
```

Comment:
❗ This line produces an error, BUG found.
To reproduce the bug: 
1. load the page
2. immediately click FIND OUT
=> ERROR in console & nothing happens in the game
![image](https://user-images.githubusercontent.com/3659568/107962567-66367a80-6fa7-11eb-8863-f5cffb4bde62.png)

**Why this is happening:**
The reason this happens is that if the user hasn't interacted with the select question, and the variable _currentQuestion_ has no value and is therefore undefined. 

**Suggested solution**
There are multiple ways to fix this, but the easiest way is to make sure the variable _currentQuestion_ has predefined values (since that is the default set option) in the start function. 

```js
const start = () => {
  // in this function add something like this:
  currentQuestion = {
    attribute: 'hairColor',
    value: 'brown',
    category: 'hair color'
  }
}
```

Reply:
Hm, I can't make it work. :/